packages/ui/src/elements/Table/RelationshipProvider/index.tsx

```diff
@@ -66,14 +67,35 @@ export const RelationshipProvider: React.FC<{ readonly children?: React.ReactNod
           params.append('depth', '0')
           params.append('limit', '250')
 
+          const collection = collections.find((c) => c.slug === slug)
+          if (collection.admin.enableListViewSelectAPI) {
+            const fieldToSelect = collection.admin.useAsTitle ?? 'id'
+            params.append(`select[${fieldToSelect}]`, 'true')
+
+            if (collection.upload) {
+              // for upload enabled collections, the FileCell component needs the following fields:
+              const fieldsToSelect = [
```

Comment:
This is fine for now, but in the future we ought to share this with `getBaseUploadFields` so that any future fields get automatically selected. And probably even some of it's logic too so that only the necessary fields are selected. For example, `focalY` only if `focalPoint` is not disabled.

Reply:
Looks like @JarrodMFlesch just created a very similar function in his PR here: https://github.com/payloadcms/payload/pull/13864. Might be worth looking into how we might deduplicate some of this logic.